R/stat-density.r

```diff
@@ -134,3 +148,26 @@ compute_density <- function(x, w, from, to, bw = "nrd0", adjust = 1,
     n = nx
   ), n = length(dens$x))
 }
+
+reflect_density <- function(dens, bounds, from, to) {
+  if (all(is.infinite(bounds))) {
+    return(dens)
+  }
+
+  f_dens <- stats::approxfun(
+    x = dens$x, y = dens$y, method = "linear", yleft = 0, yright = 0
+  )
+
+  out_x <- intersection_grid(dens$x, bounds, from, to)
+  out_y <- f_dens(out_x) + f_dens(bounds[1] + (bounds[1] - out_x)) +
+    f_dens(bounds[2] + (bounds[2] - out_x))
+
+  list(x = out_x, y = out_y)
+}
+
+intersection_grid <- function(grid, bounds, from, to) {
```

Comment:
Please add some comments explaining what the function does and what the parameters are.